apps/dashboard/src/components/in-app-action-dropdown.tsx

```diff
@@ -0,0 +1,220 @@
+import { EditorView } from '@uiw/react-codemirror';
+import { ComponentProps, useMemo } from 'react';
+import { useFormContext, useWatch } from 'react-hook-form';
+import { RiEdit2Line, RiExpandUpDownLine, RiForbid2Line } from 'react-icons/ri';
+import merge from 'lodash.merge';
+
+import { Button, buttonVariants } from '@/components/primitives/button';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@/components/primitives/dropdown-menu';
+import { Editor } from '@/components/primitives/editor';
+import {
+  FormControl,
+  FormField,
+  FormItem,
+  FormLabel,
+  FormMessage,
+  FormMessagePure,
+} from '@/components/primitives/form/form';
+import { InputField } from '@/components/primitives/input';
+import { Popover, PopoverContent, PopoverTrigger } from '@/components/primitives/popover';
+import { Separator } from '@/components/primitives/separator';
+import { URLInput } from '@/components/workflow-editor/url-input';
+import { completions } from '@/utils/liquid-autocomplete';
+import { parseStepVariablesToLiquidVariables } from '@/utils/parseStepVariablesToLiquidVariables';
+import { cn } from '@/utils/ui';
+import { urlTargetTypes } from '@/utils/url';
+import { autocompletion } from '@codemirror/autocomplete';
+import { useStep } from '@/components/workflow-editor/steps/step-provider';
+
+const primaryActionKey = 'primaryAction';
+const secondaryActionKey = 'secondaryAction';
+
+export const InAppActionDropdown = ({ onMenuItemClick }: { onMenuItemClick?: () => void }) => {
```

Comment:
This component is now "more" reusable and we use it under the `/onboarding/inbox` route.